### PR TITLE
feat(RHINENG-10818): Add connected status column to SystemTable

### PR DESCRIPTION
**Pull request by mhuth**
Before:
Screenshot from 2024-07-02 17-42-10

After:
Screenshot from 2024-07-02 17-41-53

### CODE REVIEW

**Comment by mhuth** [comment]
@johnsonm325 Hopefully this implementation is sufficient ... for now.

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (4.7.3) is older than required version (6.0.0)

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (4.7.3) is older than required version (6.0.0)

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (4.7.3) is older than required version (6.0.0)

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (4.7.3) is older than required version (6.0.0)

**Comment by PaulWay** [comment]
Just as an addendum to this:

In order to get the `connected` status, we have to request it from Playbook Dispatcher.  This is moderately expensive, at over half a second per call, so we only do this on the page of systems we are now displaying.  We get the page of systems, we request the connection status for them and only them, we add the `connected` field value to each one, and then we pass that to be rendered into JSON.

Because it is not a database field, we can't sort it and we can't filter on it.  We don't know what the list of currently connected systems is until we ask Playbook Dispatcher for the connection status of the given list of hosts.  And it's going to slow requests down A LOT if we have to get the connection status for ALL hosts just so we can sort the connected hosts up to the top of a page of ten hosts...

**Review by johnsonm325 — CHANGES_REQUESTED** [review]
We'll need to wait for UX to mock-up this feature. Once we have that we can fine-tune this PR.

Thanks!

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (4.7.3) is older than required version (6.0.0)

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (4.7.3) is older than required version (6.0.0)

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (4.7.3) is older than required version (6.0.0)

**Comment by mhuth** [comment]
Changes to the column icons, popover, width, etc:

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (4.7.3) is older than required version (6.0.0)

**Comment by mhuth** [comment]
I've updated the wording slightly to more accurately represent what is this information means to a customer:
Screenshot from 2024-07-11 11-08-24

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (4.7.3) is older than required version (6.0.0)

File: src/helpers.js [hunk 1]
```diff
@@ -25,3 +26,51 @@ export const createEligibilityTooltip = (eligibility) => {
...
+  return connected ? (
+    <span>
+      <ConnectedIcon className={xsIcon} /> Connected
+    </span>
+  ) : (
+    <Popover
+      triggerAction="hover"
+      headerContent={
+        <Title headingLevel="h4">
+          <DisconnectedIcon className={xsIcon} /> System not connected via RHC
```
Comment (ragevou) [inline src/helpers.js:42]:
```suggestion
          <DisconnectedIcon className={xsIcon} /> System is not connected via RHC
```

File: src/helpers.js [hunk 2]
```diff
@@ -25,3 +26,51 @@ export const createEligibilityTooltip = (eligibility) => {
...
+        </Title>
+      }
+      position="right"
+      bodyContent={
+        <Flex
+          direction={{ default: 'column' }}
+          spaceItems={{ default: 'spaceItemsNone' }}
+        >
+          <p>
+            Run <span className={boldText}>rhc status</span> on this system to
```
Comment (ragevou) [inline src/helpers.js:52]:
```suggestion
            Run <span className={boldText}>rhc status</span> to check if this system
```

File: src/helpers.js [hunk 3]
```diff
@@ -25,3 +26,51 @@ export const createEligibilityTooltip = (eligibility) => {
...
+            check its RHC connection to Red Hat Insights. After verifying the
```
Comment (ragevou) [inline src/helpers.js:53]:
```suggestion
            has RHC connection to Red Hat Insights. After verifying the
```

File: src/helpers.js [hunk 4]
```diff
@@ -25,3 +26,51 @@ export const createEligibilityTooltip = (eligibility) => {
...
+            connection is active, click on the{' '}
```
Comment (ragevou) [inline src/helpers.js:54]:
```suggestion
            connection is active, click the{' '}
```

**Comment by ragevou** [comment]
hi @mhuth , I left a few suggestions to improve the microcopies. Feel free to use them. cheers! =)

File: src/helpers.js (hunk 1 above)
Comment (mhuth) [reply src/helpers.js:42]: Done!

File: src/helpers.js (hunk 3 above)
Comment (mhuth) [reply src/helpers.js:53]: Done!

File: src/helpers.js (hunk 4 above)
Comment (mhuth) [reply src/helpers.js:54]: Done!

**Comment by InsightsDroid** [comment]
## :exclamation: Outdated dependencies error
* Installed version of @redhat-cloud-services/frontend-components-config (4.7.3) is older than required version (6.0.0)

**Pull request merged**